components/calendar/src/calendar_arithmetic.rs

```diff
@@ -115,10 +116,11 @@ impl<C: CalendarArithmetic> ArithmeticDate<C> {
             }
         }
 
+        #[allow(clippy::unwrap_used)] // The day is withing the month_days bound which is u8
```

Comment:
please add a debug_assert that checks this condition

Reply:
Also you don't need the `allow` since you're using `unwrap_or`, which isn't panicky

Reply:
Added one, not totally sure if that is good enough

Reply:
No, assert that `day <= C::month_days(year, month)`.

And document the function by saying that `year_day` is expected to be in range for calendar `C`

Reply:
The while loop condition already checks that right?!
The try_into() here tries to convert the i32 -> u8, so I was checking that specific case. 

Let me know if it still makes sense to add an assert_debug with month_days

Reply:
Done